### PR TITLE
i2 - FixItemControl 

### PR DESCRIPTION
**Pull request by haynescrossley**
@elliotmitchum 
FixItemControl Control class ready for review before merging into master.

Updated branch with latest master 17/8

### CODE REVIEW

**Comment by elliotmitchum** [comment]
@amrmosit This is ready for your review

File: src/library/fixitem/FixItemControl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    public FixItemControl() {
+        this.library = Library.getInstance();
+        state = ControlState.INITIALISED;
+    }
+    
+    
+    public void setUI(FixItemUI ui) {
+        if (!state.equals(ControlState.INITIALISED))
+            throw new RuntimeException("FixItemControl: cannot call setUI except in INITIALISED state");
```
Comment (amrmosit) [inline src/library/fixitem/FixItemControl.java:23]: PR 46 D1 CSG#7 Curly brackets must be in use.

File: src/library/fixitem/FixItemControl.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            
+        this.ui = ui;
+        this.ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void itemScanned(long itemId) {
+        if (!state.equals(ControlState.READY))
+            throw new RuntimeException("FixItemControl: cannot call itemScanned except in READY state");
```
Comment (amrmosit) [inline src/library/fixitem/FixItemControl.java:34]: PR 46 D2 CSG#7 Curly brackets must be used.

**Review by amrmosit — CHANGES_REQUESTED** [review]
There are two defects to be fixed. @haynescrossley @elliotmitchum

File: src/library/fixitem/FixItemControl.java (hunk 1 above)
Comment (haynescrossley) [reply src/library/fixitem/FixItemControl.java:23]: [37e8a84](https://github.com/elliotmitchum/ITC205/pull/46/commits/37e8a84c403454296d1bdb09cc9ca7b54552cde0)

File: src/library/fixitem/FixItemControl.java (hunk 2 above)
Comment (haynescrossley) [reply src/library/fixitem/FixItemControl.java:34]: [efb76bd](https://github.com/elliotmitchum/ITC205/pull/46/commits/efb76bdba38c1481c2c5f01f19f5323ccc95d3c6)

**Comment by haynescrossley** [comment]
@amrmosit changes resolved

**Review by amrmosit — APPROVED** [review]
Review is done and branch is ready to merge 
@elliotmitchum @haynescrossley

**Pull request merged**